.github/workflows/pr-check-links.yml

```diff
@@ -15,11 +15,26 @@ jobs:
           ref: ${{ github.event.pull_request.head.ref }}
           repository: ${{ github.event.pull_request.head.repo.full_name }}
 
-      - name: Check out main branch
-        run: git checkout main
+      - name: Set environment variables
+        id: set-env
+        run: |
+          # Extract the base commit ID where the feature branch diverged from main
+          base_commit=$(git merge-base origin/main ${{ github.event.pull_request.head.ref }})
```

Comment:
There is another option to find the base commit:
```
git merge-base --fork-point origin/main
```
However, be aware that `git merge-base --fork-point origin/main` considers the currently checked-out branch when determining the fork point.

Reply:
Probably, it would be safer to explicitly specify the feature branch.

I also tried to use ${{ github.event.pull_request.base.ref }} and it [does not return](https://github.com/orgs/community/discussions/39880) the correct commit ID.